client/ayon_core/plugins/publish/cleanup.py

```diff
@@ -84,6 +84,14 @@ def process(self, instance):
                 f"type is excluded from cleanup: {product_type}")
             return
 
+        explicit_cleanup_dirs = instance.data.get("explicit_cleanup_dirs", [])
+        for _dir in explicit_cleanup_dirs:
+            self.log.debug("Removing explicit cleanup dir {}".format(_dir))
+            try:
+                shutil.rmtree(_dir)
```

Comment:
Although I can see the use case - I'm a bit worried that some day, accidentally a wrong path ends up in `explicit_cleanup_dirs` :') and it ends up wiping a full project or asset folder. :)

Additionally, I feel like it should _not be_ part of this plug-in. As far as I can see this plug-in is focused on handling `stagingDir` - and explicitly does not apply to _other files_. This logic is better kept in either a separate plug-in, or actually part of the explicit cleanup (even though that is `ContextPlugin` it can iterate over the instances just fine.)